app/packages/looker-3d/src/renderables/pointcloud/shaders.tsx

```diff
@@ -132,7 +135,8 @@ const ShadeByIntensityShaders = {
   void main() {
     vUv = uv;
     vec3 pos = position;
-    hValue = remap(min, max, color.r);
+    float effectiveIntensity = useIntensity ? intensity : color.r;
+    hValue = remap(min, max, effectiveIntensity);
```

Comment:
Its been a while since I've worked on this but I believe the `hValue` mapping will changed when `useIntensity` is `true`. I think the `min` and `max` values will need to be derived in one way or another from the intensity range (possibly `(0, 1)` ?).

Reply:
To clarify `min` and `max` should probably be renamed `colorMin` and `colorMax` which would be unrelated to the `intensity` min/max.

Reply:
In general, it's common for `intensity` values in PCD files to be mapped to an 8-bit integer range, typically ranging from 0 to 255. However, to accommodate potential exceptions and ensure broader compatibility, I have utilized the `setColorMinMax(computeMinMaxForColorBufferAttribute(intensityAttribute))` in my code. This function dynamically determines the range of intensity values, which is then passed to the ShadeByIntensityShaders. This method ensures that our shading logic remains flexible and accurate, regardless of the specific range of intensity values encountered in different PCD files.

Reply:
Awesome that looks right to me!